core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/ComponentURI.java

```diff
@@ -113,6 +113,10 @@ private ComponentURI(CodeSystemURI codeSystem, short terminologyComponentId, Str
 		this.identifier = identifier;
 	}
 	
+	public static ComponentURI of(CodeSystemURI codeSystemURI, short terminologyComponentId, String identifier) {
+		return new ComponentURI(codeSystemURI, terminologyComponentId, Strings.nullToEmpty(identifier));
+	}
+	
 	@JsonCreator
 	public static ComponentURI of(String codeSystem, short terminologyComponentId, String identifier) {
```

Comment:
Rename the first argument here to `codeSystemUri` to indicate that it accepts an URI.